sympy/integrals/manualintegrate.py

```diff
@@ -733,19 +733,62 @@ def quadratic_denom_rule(integral):
     a = sympy.Wild('a', exclude=[symbol])
     b = sympy.Wild('b', exclude=[symbol])
     c = sympy.Wild('c', exclude=[symbol])
+
     match = integrand.match(a / (b * symbol ** 2 + c))
 
-    if not match:
-        return
+    if match:
+        a, b, c = match[a], match[b], match[c]
+        if b.is_extended_real and c.is_extended_real:
+            return PiecewiseRule([(ArctanRule(a, b, c, integrand, symbol), sympy.Gt(c / b, 0)),
+                                (ArccothRule(a, b, c, integrand, symbol), sympy.And(sympy.Gt(symbol ** 2, -c / b), sympy.Lt(c / b, 0))),
+                                (ArctanhRule(a, b, c, integrand, symbol), sympy.And(sympy.Lt(symbol ** 2, -c / b), sympy.Lt(c / b, 0))),
+            ], integrand, symbol)
+        else:
+            return ArctanRule(a, b, c, integrand, symbol)
 
-    a, b, c = match[a], match[b], match[c]
-    if b.is_extended_real and c.is_extended_real:
-        return PiecewiseRule([(ArctanRule(a, b, c, integrand, symbol), sympy.Gt(c / b, 0)),
-                              (ArccothRule(a, b, c, integrand, symbol), sympy.And(sympy.Gt(symbol ** 2, -c / b), sympy.Lt(c / b, 0))),
-                              (ArctanhRule(a, b, c, integrand, symbol), sympy.And(sympy.Lt(symbol ** 2, -c / b), sympy.Lt(c / b, 0))),
-        ], integrand, symbol)
-    else:
-        return ArctanRule(a, b, c, integrand, symbol)
+    d = sympy.Wild('d', exclude=[symbol])
+    match2 = integrand.match(a / (b * symbol ** 2 + c * symbol + d))
+    if match2:
+        b, c =  match2[b], match2[c]
+        if b.is_zero:
+            return
+        u = sympy.Dummy('u')
+        u_func = symbol + c/(2*b)
+        integrand2 = integrand.subs(symbol, u - c / (2*b))
+        next_step = integral_steps(integrand2, u)
+        if next_step:
+            return URule(u, u_func, None, next_step, integrand2, symbol)
```

Comment:
What will happen if there is no `next_step`. Now `b` and `c` are not wild symbols so `match3` may not work as expected.

Reply:
I don't think that is possible, but just to make sure, I added a line to handle that case.